node/chainSimulator/components/bootstrapComponents_test.go

```diff
@@ -0,0 +1,191 @@
+				return &testscommon.RaterMock{}
+			},
+			NodesShufflerCalled: func() nodesCoordinator.NodesShuffler {
+				return &shardingMocks.NodeShufflerMock{}
+			},
+			RoundHandlerCalled: func() consensus.RoundHandler {
+				return &testscommon.RoundHandlerMock{}
+			},
+			HasherCalled: func() hashing.Hasher {
+				return &testscommon.HasherStub{}
+			},
+			PathHandlerCalled: func() storage.PathManagerHandler {
+				return &testscommon.PathManagerStub{}
+			},
+			TxMarshalizerCalled: func() marshal.Marshalizer {
+				return &testscommon.MarshallerStub{}
+			},
+			AddressPubKeyConverterCalled: func() core.PubkeyConverter {
+				return &testscommon.PubkeyConverterStub{}
+			},
+			Uint64ByteSliceConverterCalled: func() typeConverters.Uint64ByteSliceConverter {
+				return &mock.Uint64ByteSliceConverterMock{}
+			},
+			TxSignHasherCalled: func() hashing.Hasher {
+				return &testscommon.HasherStub{}
+			},
+		},
+		CryptoComponents: &mock.CryptoComponentsStub{
+			PubKey:                  &mock.PublicKeyMock{},
+			BlockSig:                &cryptoMocks.SingleSignerStub{},
+			BlKeyGen:                &cryptoMocks.KeyGenStub{},
+			TxSig:                   &cryptoMocks.SingleSignerStub{},
+			TxKeyGen:                &cryptoMocks.KeyGenStub{},
+			ManagedPeersHolderField: &testscommon.ManagedPeersHolderStub{},
+		},
+		NetworkComponents: &mock.NetworkComponentsStub{
+			Messenger:                        &p2pmocks.MessengerStub{},
+			FullArchiveNetworkMessengerField: &p2pmocks.MessengerStub{},
+		},
+		StatusCoreComponents: &factory.StatusCoreComponentsStub{
+			TrieSyncStatisticsField: &testscommon.SizeSyncStatisticsHandlerStub{},
+			AppStatusHandlerField:   &statusHandler.AppStatusHandlerStub{},
+			StateStatsHandlerField:  &testscommon.StateStatisticsHandlerStub{},
+		},
+		WorkingDir:     ".",
+		FlagsConfig:    config.ContextFlagsConfig{},
+		ImportDBConfig: config.ImportDbConfig{},
+		PrefsConfig:    config.Preferences{},
+		Config: config.Config{
+			EpochStartConfig: config.EpochStartConfig{
+				MinNumConnectedPeersToStart:       1,
+				MinNumOfPeersToConsiderBlockValid: 1,
+			},
+			TrieSync: config.TrieSyncConfig{
+				MaxHardCapForMissingNodes: 1,
+				NumConcurrentTrieSyncers:  1,
+			},
+			GeneralSettings: config.GeneralSettingsConfig{
+				SetGuardianEpochsDelay: 1,
+			},
+			Versions: config.VersionsConfig{
+				Cache: config.CacheConfig{
+					Type:     "LRU",
+					Capacity: 123,
+				},
+				DefaultVersion: "1",
+				VersionsByEpochs: []config.VersionByEpochs{
+					{
+						StartEpoch: 0,
+						Version:    "1",
+					},
+				},
+			},
+			WhiteListPool: config.CacheConfig{
+				Type:     "LRU",
+				Capacity: 123,
+			},
+		},
+		ShardIDStr: "0",
+	}
+}
+
+func TestCreateBootstrapComponents(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		comp, err := CreateBootstrapComponents(createArgsBootstrapComponentsHolder())
```

Comment:
should we call a comp.Close() at the end of the test?

Reply:
nope, added for coverage only

Reply:
added everywhere